ValueTypeHelpers.cpp:181:18: error: unused variable 'clazz'

internal build job/Build_JDK16_x86-64_linux_Personal/92/

Due to https://github.com/eclipse/openj9/pull/12425

```
14:50:57  /home/jenkins/workspace/Build_JDK16_x86-64_linux_Personal/openj9/runtime/vm/ValueTypeHelpers.cpp: In function 'BOOLEAN isFlattenableFieldFlattened(J9Class*, J9ROMFieldShape*)':
14:50:57  /home/jenkins/workspace/Build_JDK16_x86-64_linux_Personal/openj9/runtime/vm/ValueTypeHelpers.cpp:181:18: error: unused variable 'clazz' [-Werror=unused-variable]
14:50:57           J9Class* clazz = getFlattenableFieldType(fieldOwner, field);
14:50:57                    ^~~~~
14:50:57  cc1plus: all warnings being treated as errors
14:50:57  runtime/vm/CMakeFiles/j9vm.dir/build.make:1177: recipe for target 'runtime/vm/CMakeFiles/j9vm.dir/ValueTypeHelpers.cpp.o' failed
```

@hangshao0 @gacholio fyi

--- a/runtime/vm/ValueTypeHelpers.cpp
+++ b/runtime/vm/ValueTypeHelpers.cpp
@@ -177,9 +177,7 @@ isFlattenableFieldFlattened(J9Class *fieldOwner, J9ROMFieldShape *field)
 {
         Assert_VM_notNull(fieldOwner);
         Assert_VM_notNull(field);
-
-        J9Class* clazz = getFlattenableFieldType(fieldOwner, field);
-        BOOLEAN fieldFlattened = J9_IS_FIELD_FLATTENED(clazz, field);
+        BOOLEAN fieldFlattened = J9_IS_FIELD_FLATTENED(getFlattenableFieldType(fieldOwner, field), field);
 
         return fieldFlattened;
 }

